src/plugins/explore/public/application/pages/traces/trace_details/public/logs/logs_data_table.tsx

```diff
@@ -0,0 +1,194 @@
+          defaultMessage: 'Message',
+        }),
+        sortable: false,
+        render: (message: string) => (
+          <EuiText size="xs" style={{ wordBreak: 'break-word', whiteSpace: 'pre-wrap' }}>
+            {message || 'N/A'}
+          </EuiText>
+        ),
+      },
+      {
+        field: 'spanId',
+        name: i18n.translate('explore.logsDataTable.column.spanId', {
+          defaultMessage: 'Span ID',
+        }),
+        sortable: false,
+        width: '200px',
+        render: (spanId: string) => {
+          if (!spanId) return <span>-</span>;
+          return (
+            <EuiText
+              size="xs"
+              style={{
+                fontFamily: 'monospace',
+                wordBreak: 'break-all',
+                color: onSpanClick ? '#006BB4' : 'inherit',
+                cursor: onSpanClick ? 'pointer' : 'default',
+                textDecoration: onSpanClick ? 'underline' : 'none',
+              }}
+              onClick={() => onSpanClick && onSpanClick(spanId)}
+            >
+              {spanId}
+            </EuiText>
+          );
+        },
+      },
+    ];
+
+    if (compactMode) {
+      return allColumns.filter((column) => column.field === 'message');
+    }
+
+    return allColumns;
+  }, [onSpanClick, compactMode]);
+
+  const sortedLogs = useMemo(() => {
+    // In compact mode, show all logs without sorting or pagination
+    if (compactMode) {
+      return logs;
+    }
+
+    const sorted = [...logs].sort((a, b) => {
+      let aValue: string | number;
+      let bValue: string | number;
+
+      if (sortField === 'timestamp') {
+        aValue = a.timestamp;
+        bValue = b.timestamp;
+      } else if (sortField === 'level') {
+        aValue = a.level || '';
+        bValue = b.level || '';
+      } else {
+        aValue = '';
+        bValue = '';
+      }
+
+      if (aValue < bValue) {
+        return sortDirection === 'asc' ? -1 : 1;
+      }
+      if (aValue > bValue) {
+        return sortDirection === 'asc' ? 1 : -1;
+      }
+      return 0;
+    });
+
+    const startIndex = pageIndex * pageSize;
+    return sorted.slice(startIndex, startIndex + pageSize);
+  }, [logs, sortField, sortDirection, pageIndex, pageSize, compactMode]);
+
+  const pagination = useMemo(
+    () => ({
+      pageIndex,
+      pageSize,
+      totalItemCount: logs.length,
+      pageSizeOptions: [10, 25, 50, 100],
+    }),
+    [pageIndex, pageSize, logs.length]
+  );
+
+  const onTableChange = ({ page, sort }: TableChangeParams) => {
+    if (page) {
+      setPageIndex(page.index);
+      setPageSize(page.size);
+    }
+    if (sort) {
+      setSortField(sort.field);
+      setSortDirection(sort.direction);
+    }
+  };
+
+  if (isLoading) {
+    return <div>Loading logs...</div>;
+  }
+
+  const tableProps = {
+    items: sortedLogs,
+    columns,
+    loading: isLoading,
+    tableLayout: 'auto' as const,
+    'data-test-subj': 'logs-data-table',
```

Comment:
should we do something like `exploreTracesLogsDataTable` or something? I don't know what prefix convention we have been following for Traces, but we should prefix this data-test-subj